.github/workflows/test-external-providers.yml

```diff
@@ -0,0 +1,89 @@
+name: Test External Providers
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test-external-providers:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install Ollama
```

Comment:
we could probably spin ollama stuff into an action that we could re-use everywhere.

Reply:
yeah at one point we should!